## Case Study on Measures of Central Tendency and Dispersion

### Importing required libraries

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import os

### Reading the dataset into Python

In [3]:
dataset1=pd.read_csv('Desktop\DSA\StudentsPerformance1.csv')
dataset1

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [5]:
dataset1.isnull().sum()

gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

###   1. To find out the number of males and females participated in the test.

In [9]:
count1=dataset1['gender'].count()
print("No of Males and Females in the data set: " , count1)

No of Males and Females in the data set:  1000


In [6]:
gender_count=dataset1['gender'].unique()
print('The number of Males and females in the gender are as follows : ')
dataset1.groupby(['gender']).count()

The number of Males and females in the gender are as follows : 


race/ethnicity  parental level of education  lunch  \
gender                                                       
female             518                          518    518   
male               482                          482    482   

        test preparation course  math score  reading score  writing score  
gender                                                                     
female                      518         518            518            518  
male                        482         482            482            482

In [17]:
dataset1["gender"].value_counts()

female    518
male      482
Name: gender, dtype: int64

##### Insights: There are more number of females than the males.

### 2. Students Parental level of Education

In [16]:
AvgP=dataset1.groupby("parental level of education").mean()
AvgP

math score  reading score  writing score
parental level of education                                          
associate's degree            67.977477      71.018018      70.031532
bachelor's degree             69.288136      73.000000      73.381356
high school                   61.821429      64.602041      62.403061
master's degree               70.254237      75.949153      75.677966
some college                  67.128319      69.566372      69.035398
some high school              63.134078      66.759777      64.888268

In [18]:
dataset1.groupby("parental level of education", as_index=False).size()

parental level of education  size
0          associate's degree   222
1           bachelor's degree   118
2                 high school   196
3             master's degree    59
4                some college   226
5            some high school   179

##### Insights:
From the dataset, It is clear that only few students parents have degree and master's degree level of education(around 40%) and the rest of the parents did not have a degree or master's and they are more in numbers also (i.e above 60%). Also the students who have scored above average or good marks are the children of parents with master's degree, bachelor's and associate degree. 

###  3. Who scores the most on average for math, reading and writing based on 1) Gender and 2) Test preparation Course


In [19]:
Avg1=dataset1.groupby("gender").mean()
Avg1


math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

##### Insights:
From the analysis, the males have scored maximum in the Maths score than the females in the dataset whereas the females scored more in both reading and writing than the males in the dataset.

In [20]:
Avg2=dataset1.groupby("test preparation course").mean()
Avg2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

#### Insights:
Those who have completed the test preparation course have scored more than those who have not yet completed.

### 4. What do you think about the scoring variation for math, reading and writing based on 1)Gender and 2) Test Preparation Course.




In [21]:
dataset1.loc[:,['math score','reading score','writing score']].std()

math score       15.569567
reading score    14.706436
writing score    15.226381
dtype: float64

In [23]:
st1=dataset1.groupby("gender").std()
st1

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

##### Insights:
Males show less deviation than the  females in the dataset in the math score. But they both show almost same deviation in writing and reading scores.

In [24]:
st2=dataset1.groupby("test preparation course").std()
st2

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

##### Insights:
From this data analysed, it is shown that those who completed the test preparation course shows less deviation than those who have not completed yet.



### 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [25]:
dataset1.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000

In [39]:
dataset1["math score"].quantile(0.25)

57.0

In [40]:
np.percentile(dataset1['math score'] , [25,50,75])

array([57., 66., 77.])

In [41]:
pd.qcut(dataset1['math score'], q=4).value_counts()

(66.0, 77.0]      266
(-0.001, 57.0]    265
(57.0, 66.0]      238
(77.0, 100.0]     231
Name: math score, dtype: int64

#### Insights:
From the above describing function we have obtained different statistical data of the dataset given and we can see the maths score analysis, above 25% of the students have scored 75% i.e more than 77 marks in maths score.
For 231 students, the management should give bonus points.